modules/test/playwright/pages/calendar-web/CalendarWidgetPage.ts

```diff
@@ -50,6 +52,14 @@ export class CalendarWidgetPage {
 			.frameLocator('iframe')
 			.getByLabel('Ends', {exact: true});
 		this.modalRecurrencePage = new ModalRecurrencePage(page);
+		this.miniCalendarBase = page.locator('.yui3-calendarbase');
+		this.miniCalendarGrid = page.locator('.yui3-calendar-grid');
+		this.miniCalendarHeaderLabel = page.locator(
+			'.yui3-calendar-header-label'
+		);
```

Comment:
For these, We could also get them by their roles, like bellow, and it would also work, since at this moment, each element has unique roles, but I think it's too general and we would risk the future possibility of having another element in the page that would share one of these roles, so in my opinion it would be better to leave it like that, using the classes, unless we can make this more specific. Let me know if you think otherwise, and feel free to handle this as you see fit! 

```
this.miniCalendarBase = page.getByRole('dialog');
this.miniCalendarGrid = page.getByRole('grid');
this.miniCalendarHeaderLabel = page.getByRole('paragraph');
```

Reply:
Regarding these locators, I believe that using classes will bring us greater flexibility of use. As locators must be generic enough to be used in all tests, I think we can find more than one occurrence for these locators with some ease.

Even more so considering that this calendar widget will be used within a page, often containing other elements.

But great observation, that's the spirit of the thing!